From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benjamin Marzinski <bmarzins@redhat.com>
Date: Mon, 23 Apr 2018 17:53:02 -0500
Subject: [PATCH] multipathd: fix reservation_key check

mpath_pr_event_handle should exit if the reservation key is zero, not
non-zero.

Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 multipathd/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/multipathd/main.c b/multipathd/main.c
index 841d3e9..84af946 100644
--- a/multipathd/main.c
+++ b/multipathd/main.c
@@ -3136,7 +3136,7 @@ int mpath_pr_event_handle(struct path *pp)
 
 	mpp = pp->mpp;
 
-	if (get_be64(mpp->reservation_key))
+	if (!get_be64(mpp->reservation_key))
 		return -1;
 
 	pthread_attr_init(&attr);
-- 
2.7.4

